sql/descriptor.go

```diff
@@ -62,12 +62,15 @@ type descriptorProto interface {
 }
 
 // checkPrivilege verifies that p.user has `privilege` on `descriptor`.
-func (p *planner) checkPrivilege(descriptor descriptorProto, privilege privilege.Kind) error {
-	if descriptor.GetPrivileges().CheckPrivilege(p.user, privilege) {
+func (p *planner) checkPrivilege(descriptor descriptorProto, priv privilege.Kind) error {
+	if descriptor.GetPrivileges().CheckPrivilege(p.user, priv) {
 		return nil
 	}
+	if priv != privilege.SELECT && isSystemConfigID(descriptor.GetID()) {
+		return fmt.Errorf("system database is read-only")
```

Comment:
I think we'll definitely want to cover the system db in the docs. We'll want to at least mention:
* what each table is and what useful information can be gotten out of them
* what the special privileges are and why (eg: we don't want you to drop a table, that would break everything)